### PR TITLE
Homework #1

### PR DESCRIPTION
**Pull request by vdumbrav**
**Boilerplate from** https://github.com/tr1s/tris-webpack-boilerplate

Work in IE 11

### CODE REVIEW

File: src/index.scss [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+  &__list {
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    padding: 0;
+    margin: 0;
+  }
+
+  &__item {
+    padding: 10px 0;
```
Comment (IvanCeban) [inline src/index.scss]: there is no sense to add padding to header__item elements. as you position them vertically on center and they always stay aligned horizontally.

File: src/index.scss [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    list-style: none;
+  }
+
+  &__link {
+    padding: 0 10px;
```
Comment (IvanCeban) [inline src/index.scss]: More semantic way will be to add margin-left to all header__item elements, except the first one. In this case you will avoid extra spacing on the first element, and to add space between elements better use 'margin'. 'padding' is used for spacing inside an element.

File: src/index.scss [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    text-decoration: none;
+
+    &:hover {
+      text-decoration: underline;
+      cursor: pointer;
+    }
+  }
+}
+
+.sidebar {
```
Comment (IvanCeban) [inline src/index.scss:96]: Same suggestions as for header

File: src/index.html [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            </li>
+            <li class="header__item">
+              <a class="header__link" href="#">gallery</a>
+            </li>
+          </ul>
+        </nav>
+        <a class="header__link" href="#">login</a>
+      </header>
+      <aside class="containter__sidebar sidebar">
+        <p class="sidebar__menu">menu</p>
```
Comment (IvanCeban) [inline src/index.html]: Better use `<span>` as it is not a paragraph

File: src/index.html [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        </div>
+        <div class="content__three">
+          <span>content 3</span>
+        </div>
+        <div class="content__four">
+          <span>content 4</span>
+        </div>
+      </main>
+      <footer class="containter__footer">
+        <p class="containter__footer-title">footer</p>
```
Comment (IvanCeban) [inline src/index.html]: Better use `<span>` as it is not a paragraph

File: src/index.scss [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+  font-weight: 600;
+
+  &__logo {
+    grid-area: logo;
+    background: #96ceb4;
+
+    &-title {
+      @include center;
+
+      text-transform: uppercase;
```
Comment (IvanCeban) [inline src/index.scss]: As we see, you have just 1 place where text is Capitalized (Menu nav). In all other places the text is UPPERCASE. It makes sense to set the text-transform: uppercase; globally, to body for example. And set it Capitalized to menu nav only. In this case you will save 3 lines of code.

File: src/index.scss [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+    'three  three  four';
+  grid-template-rows: 1fr 1fr;
+  grid-template-columns: 1fr 3fr 2fr;
+  grid-gap: 10px;
+  padding: 30px;
+  text-align: center;
+  text-transform: uppercase;
+
+  &__one {
+    box-sizing: border-box;
```
Comment (IvanCeban) [inline src/index.scss:144]: better  box-sizing: border-box; to body. Will be useful to have this property on all elements.

File: src/index.scss [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+    grid-area: one;
+    padding: 5px;
```
Comment (IvanCeban) [inline src/index.scss:146]: As I understand, you added the padding to make it look better when the screen gets smaller. My suggestion is not important for current solution, but in general better add same padding for all .content children.

**Review by IvanCeban — COMMENTED** [review]
Thanks for using Webpack, SCSS, BEM and HTML5 tags. You made me happy :)
There are some minor suggestions, but you did really good job!

File: src/index.scss [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+      'four';
+    grid-template-rows: 1fr 2fr 3fr 4fr;
+    grid-template-columns: 100%;
+    padding: 20px;
+  }
+}
+
+@media (max-width: 580px) {
+  .containter {
+    display: grid;
```
Comment (IvanCeban) [inline src/index.scss]: You already set up the display:  grid; property. There is no sense to rewrite it again for smaller screens.

File: src/index.scss [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+  }
+}
+
+@media (max-width: 767px) {
+  .containter {
+    font-size: 18px;
+  }
+
+  .content {
+    display: grid;
```
Comment (IvanCeban) [inline src/index.scss]: You already set up the display:  grid; property. There is no sense to rewrite it again for smaller screens.

File: src/index.scss (hunk 6 above)
Comment (vdumbrav) [reply src/index.scss]: ok

File: src/index.scss (hunk 7 above)
Comment (vdumbrav) [reply src/index.scss:144]: ok

File: src/index.scss (hunk 9 above)
Comment (vdumbrav) [reply src/index.scss]: ok

File: src/index.scss (hunk 10 above)
Comment (vdumbrav) [reply src/index.scss]: ok

File: src/index.scss (hunk 1 above)
Comment (vdumbrav) [reply src/index.scss]: ok

File: src/index.scss [hunk 11]
```diff
@@ -0,0 +1,227 @@
...
+}
+
+.containter {
+  display: grid;
+  grid-template-areas:
+    'logo     header'
+    'sidebar  content'
+    'sidebar  content'
+    'footer   footer';
+  grid-template-rows: 50px minmax(calc(50vh - 75px), 100%) minmax(calc(50vh - 75px), 100%) 50px;
```
Comment (IvanCeban) [inline src/index.scss:21]: Try to replace minmax(calc(50vh - 75px), 100%) with auto. It should work too. But with less code
50px auto auto 50px

File: src/index.scss (hunk 3 above)
Comment (vdumbrav) [reply src/index.scss:96]: I will change only padding to margin

File: src/index.scss (hunk 11 above)
Comment (vdumbrav) [reply src/index.scss:21]: Yes, but in IE `auto` works wrong
https://css-tricks.com/css-grid-in-ie-css-grid-and-the-new-autoprefixer/#article-header-id-15

File: src/index.scss (hunk 2 above)
Comment (vdumbrav) [reply src/index.scss]: ok

File: src/index.scss (hunk 7 above)
Comment (vdumbrav) [reply src/index.scss:144]: in body not working, but use `*` I dont want

File: src/index.scss (hunk 8 above)
Comment (vdumbrav) [reply src/index.scss:146]: It is your opinion. I will not apply this recomendation )

File: src/index.scss (hunk 11 above)
Comment (IvanCeban) [reply src/index.scss:21]: Cool, nice notice. Then keep it like that

File: src/index.scss (hunk 8 above)
Comment (IvanCeban) [reply src/index.scss:146]: You're the boss ;)

File: src/index.scss (hunk 7 above)
Comment (IvanCeban) [reply src/index.scss:144]: why not using * ?

**Comment by vdumbrav** [comment]
Done.

File: src/index.scss (hunk 7 above)
Comment (vdumbrav) [reply src/index.scss:144]: I have setting in stylelint. I thought that `*` it is not good practice

File: src/index.html (hunk 4 above)
Comment (vdumbrav) [reply src/index.html]: ok

File: src/index.html (hunk 5 above)
Comment (vdumbrav) [reply src/index.html]: ok

File: src/index.scss (hunk 7 above)
Comment (IvanCeban) [reply src/index.scss:144]: From one point of view yes. But from another you have to add border-box to certain elements which is not useful at all. In this layout I completely accept your solution. But in a real project I would not agree with you.

Comment (vdumbrav) [reply src/index.scss:144]: In real project I setup normalize.scss or reset.scss

**Review by IvanCeban — APPROVED** [review]

**Pull request merged**